### ISSUE TITLE
Update core examples listings

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#10392]
|[<img alt="puzzlement's avatar" src="https://avatars.githubusercontent.com/u/94942?s=50" width="50" height="50">](https://github.com/puzzlement)| @puzzlement reported|
|-|-|
|Trac ID|trac#506|
|Type|enhancement|
|Created|2004-02-15 08:27:57Z|
|Branch|https://github.com/twisted/twisted/tree/core-examples-subdir-506-2|

<details><summary>Searchable metadata</summary>

```
trac-id__506 506
type__enhancement enhancement
reporter__hypatia hypatia
priority__low low
milestone__ 
branch__branches_core_examples_subdir_506_2 branches/core-examples-subdir-506-2
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1076833677000000 1076833677000000
changetime__1237740871000000 1237740871000000
version__ 
owner__thijs thijs
cc__spiv cc__thijs cc__exarkun
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="puzzlement's avatar" src="https://avatars.githubusercontent.com/u/94942?s=50" width="50" height="50">](https://github.com/puzzlement)|@puzzlement commented|
|-|-|

```
#!html
<pre>
The examples directory is quite large, and should probably be split into
subdirectories &amp;mdash; if nothing else we could start auto-generating the
examples index.

Obviously doing this involves changing every howto that uses an example, and the
index.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)|@radix commented|
|-|-|

```
#!html
<pre>
this has been alleviated since theh split, although doc/core/examples still has
47 entries.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="puzzlement's avatar" src="https://avatars.githubusercontent.com/u/94942?s=50" width="50" height="50">](https://github.com/puzzlement)|@puzzlement set owner to edsuom|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_4"></a>|@thijstriemstra commented|
|-|-|

not sure what needs to happen here?

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_5"></a>|@thijstriemstra set owner to @thijstriemstra<br>@thijstriemstra set status to `assigned`|
|-|-|

From today's IRC chat:

```
exarkun: thijs: There are still a lot of examples in the core examples directory.
[14:34] exarkun: thijs: If having a lot of examples in one directory is bad, then the ticket is probably still valid.
[14:34] thijs: exarkun: ok, ill go through them
[14:34] exarkun: It's too bad the ticket doesn't explain why having a lot of examples in one directory is bad, though.
[14:34] thijs: so what about adding subdirs to that examples folder
[14:35] thijs: or does it suggest to move it to other projects
[14:36] exarkun: More the former than the latter
```

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_6"></a>|@thijstriemstra commented|
|-|-|

(In [24685]) Branching to 'core-examples-subdir-506'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)|@thijstriemstra removed owner<br>@thijstriemstra set status to `new`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun set owner to @thijstriemstra|
|-|-|

Building the docs fails.  Lore docs link to howtos sometimes, and now all the paths are different.

I'd like to consider what the advantage of this change is before doing much more work on it (unless you don't mind doing the work and then hearing someone object to the whole thing :).

Are directories with 40 files in them _really_ an impediment to learning?  What problem is being solved here?

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_9"></a>|@thijstriemstra set status to `assigned`|
|-|-|

I agree exarkun, there isn't much benefit with 40 files, but later on it can become an issue. Anyway, while working on this ticket I noticed some of the samples aren't listed in the .xhtml file, perhaps I can update that and forget about the directory changes?

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_10"></a>|@thijstriemstra commented|
|-|-|

Updating ticket title to make this ticket useful.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_11"></a>|@exarkun commented|
|-|-|

Yes, please do that.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_12"></a>|@thijstriemstra commented|
|-|-|

(In [26475]) Branching to 'core-examples-subdir-506-2'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_13"></a>|@thijstriemstra removed owner<br>@thijstriemstra set status to `new`|
|-|-|

Updated the listing in r26477, [doc builder](http://buildbot.twistedmatrix.com/builders/documentation/builds/350) is happy a well, up for review.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_14"></a>|@exarkun set owner to @thijstriemstra|
|-|-|

  1. _<li><a href="rotatinglog.py">rotatinglog.py</a> - example of using the rotating log</li>_ is a bit clumsy.  I would say _example of log file rotation_.
  1. Could you add a description for ptyserv.py?  Something like _serve shells in pseudo-terminals over TCP_.
  1. There is an example.html entry, but I don't know what it points at (there is no such file).  Delete it?

Otherwise looks good.  Please merge when the above are fixed (maybe after another doc builder run).

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_15"></a>|@thijstriemstra commented|
|-|-|

(In [26501]) Address review comments refs [#10392](https://github.com/twisted/twisted/issues/10392)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_16"></a>|@thijstriemstra set status to `assigned`|
|-|-|

The docbuilder is [happy](http://buildbot.twistedmatrix.com/builders/documentation/builds/357) so merging it.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_17"></a>|@thijstriemstra set status to `closed`|
|-|-|

(In [26503]) Merge core-examples-subdir-506-2: updating the core examples index

Author: thijs
Reviewer: exarkun
Fixes: [#10392](https://github.com/twisted/twisted/issues/10392)
